apps/ios/Shared/Views/Chat/ChatView.swift

```diff
@@ -716,31 +717,53 @@ struct ChatView: View {
 
         var revealed: Bool { chatItem == revealedChatItem }
 
+        struct TimeSeparation {
+            let isTimestampShown: Bool
+            let isGapLarge: Bool
+
+            init(for chatItem: ChatItem, at index: Int?) {
```

Comment:
inits are not supposed to do any work, they are supposed to be dummy constructor.

Making inits do some logic is OOP anti-pattern